### ISSUE TITLE
[BUG]: Digital Miner Voids other Mek Machines

### ISSUE THREAD

**Issue by SpoilersMC** [Thorfusion/Mekanism-Community-Edition#133]
### Describe the bug

If a Digital Miner is configured to mine any other machine from mekanism, the block will be erased from existence
this is due to how the Digital Miner attempts to determine item drops for the target block
Digital Miner on operation calls MinerUtils.getDrops():
https://github.com/Thorfusion/Mekanism-Community-Edition/blob/1.7.10/src/main/java/mekanism/common/tile/TileEntityDigitalMiner.java#L251
MinerUtils.getDrops() calls Block.getDrops():
https://github.com/Thorfusion/Mekanism-Community-Edition/blob/1.7.10/src/main/java/mekanism/common/util/MinerUtils.java#L35
Block.getDrops() calls Block.getItemDropped(), BlockMachine overrides getItemDropped() and returns null:
https://github.com/Thorfusion/Mekanism-Community-Edition/blob/1.7.10/src/main/java/mekanism/common/block/BlockMachine.java#L754

in conclusion every single BlockMachine that a Digital Miner attempts to collect will be voided 

### To Reproduce

Configure a Digital Miner to mine any machine from Mekanism

### Expected behavior

Collect the block item correctly.

### Mekanism Version

9.10.29

### Minecraft Version is this regarding?

1.7.10

### The crash report in folder ./crash-reports (both server and client logs)

```shell
N/A
```

**Comment by SpoilersMC** [comment]
Also affects BlockBasic, BlockCardboardBox, BlockEnergyCube, BlockGasTank, and BlockGenerator
